In [1]:
import pandas as pd
import os
import statsmodels.api as sm
import numpy as np

In [2]:
filepath = os.path.join('Resources','bikecounts.csv')
bikecounts_df = pd.read_csv(filepath)
bikecounts_df['Day'] = pd.to_datetime(bikecounts_df['Day'])
bikecounts_df.head()

Unnamed: 0        Date        Day  High Temp (°F)  Low Temp (°F)  \
0           0  2016-04-01 2016-04-01            78.1           66.0   
1           1  2016-04-02 2016-04-02            55.0           48.9   
2           2  2016-04-03 2016-04-03            39.9           34.0   
3           3  2016-04-04 2016-04-04            44.1           33.1   
4           4  2016-04-05 2016-04-05            42.1           26.1   

  Precipitation  Brooklyn Bridge  Manhattan Bridge  Williamsburg Bridge  \
0          0.01           1704.0              3126               4115.0   
1          0.15            827.0              1646               2565.0   
2          0.09            526.0              1232               1695.0   
3      0.47 (S)            521.0              1067               1440.0   
4             0           1416.0              2617               3081.0   

   Queensboro Bridge  Total  
0             2552.0  11497  
1             1884.0   6922  
2             1306.0   4759  
3             1307.0   4335  
4             2357.0   9471

In [3]:
bikecounts_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   Unnamed: 0           30 non-null     int64         
 1   Date                 30 non-null     object        
 2   Day                  30 non-null     datetime64[ns]
 3   High Temp (°F)       30 non-null     float64       
 4   Low Temp (°F)        30 non-null     float64       
 5   Precipitation        30 non-null     object        
 6   Brooklyn Bridge      30 non-null     float64       
 7   Manhattan Bridge     30 non-null     int64         
 8   Williamsburg Bridge  30 non-null     float64       
 9   Queensboro Bridge    30 non-null     float64       
 10  Total                30 non-null     int64         
dtypes: datetime64[ns](1), float64(5), int64(3), object(2)
memory usage: 2.7+ KB


In [4]:
def remove_char(value):
    for c in ['T', '(S)', ' ']:
        value = value.replace(c, '')
    return value.strip()

In [5]:
bikecounts_df['Precipitation'] = bikecounts_df['Precipitation'].apply(remove_char)
bikecounts_df['Precipitation'] = bikecounts_df['Precipitation'].replace('',0).astype('float64')
bikecounts_df.head()

Unnamed: 0        Date        Day  High Temp (°F)  Low Temp (°F)  \
0           0  2016-04-01 2016-04-01            78.1           66.0   
1           1  2016-04-02 2016-04-02            55.0           48.9   
2           2  2016-04-03 2016-04-03            39.9           34.0   
3           3  2016-04-04 2016-04-04            44.1           33.1   
4           4  2016-04-05 2016-04-05            42.1           26.1   

   Precipitation  Brooklyn Bridge  Manhattan Bridge  Williamsburg Bridge  \
0           0.01           1704.0              3126               4115.0   
1           0.15            827.0              1646               2565.0   
2           0.09            526.0              1232               1695.0   
3           0.47            521.0              1067               1440.0   
4           0.00           1416.0              2617               3081.0   

   Queensboro Bridge  Total  
0             2552.0  11497  
1             1884.0   6922  
2             1306.0   4759  
3             1307.0   4335  
4             2357.0   9471

In [6]:
bikecounts_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   Unnamed: 0           30 non-null     int64         
 1   Date                 30 non-null     object        
 2   Day                  30 non-null     datetime64[ns]
 3   High Temp (°F)       30 non-null     float64       
 4   Low Temp (°F)        30 non-null     float64       
 5   Precipitation        30 non-null     float64       
 6   Brooklyn Bridge      30 non-null     float64       
 7   Manhattan Bridge     30 non-null     int64         
 8   Williamsburg Bridge  30 non-null     float64       
 9   Queensboro Bridge    30 non-null     float64       
 10  Total                30 non-null     int64         
dtypes: datetime64[ns](1), float64(6), int64(3), object(1)
memory usage: 2.7+ KB


In [7]:
bikecounts_df.head()

Unnamed: 0        Date        Day  High Temp (°F)  Low Temp (°F)  \
0           0  2016-04-01 2016-04-01            78.1           66.0   
1           1  2016-04-02 2016-04-02            55.0           48.9   
2           2  2016-04-03 2016-04-03            39.9           34.0   
3           3  2016-04-04 2016-04-04            44.1           33.1   
4           4  2016-04-05 2016-04-05            42.1           26.1   

   Precipitation  Brooklyn Bridge  Manhattan Bridge  Williamsburg Bridge  \
0           0.01           1704.0              3126               4115.0   
1           0.15            827.0              1646               2565.0   
2           0.09            526.0              1232               1695.0   
3           0.47            521.0              1067               1440.0   
4           0.00           1416.0              2617               3081.0   

   Queensboro Bridge  Total  
0             2552.0  11497  
1             1884.0   6922  
2             1306.0   4759  
3             1307.0   4335  
4             2357.0   9471

In [8]:
X = bikecounts_df["Precipitation"]
y = bikecounts_df['Brooklyn Bridge']
X = sm.add_constant(X)
model = sm.OLS(y, X)
results = model.fit()

C:\Users\Sheryl\anaconda3\lib\site-packages\statsmodels\tsa\tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


In [9]:
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:        Brooklyn Bridge   R-squared:                       0.359
Model:                            OLS   Adj. R-squared:                  0.337
Method:                 Least Squares   F-statistic:                     15.71
Date:                Fri, 22 Oct 2021   Prob (F-statistic):           0.000463
Time:                        20:29:54   Log-Likelihood:                -242.48
No. Observations:                  30   AIC:                             489.0
Df Residuals:                      28   BIC:                             491.8
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const          2571.0184    166.446     15.447

> Hypothesis: The number of bikes crossing the Brooklyn Bridge is proportional to the Precipitation.

> Null Hypothesis: The number of bikes crossing the Brooklyn Bridge is independent of Precipitation.

From the results of the OLS regression we can see that the p value is 0.000463 which is < 0.05 and hence the Null hypothesis can be rejected.

We also see that the R-squared value is 0.359 which is a positive value which indicates that there exists a direct relation between the number of bikes crossing the Brooklyn Bridge and the Precipitation.